Handle ConversionError from Schematics in SchematicsModelField

thisDash opened the pull request:
Added the catching of ConversionError from schematics in the convert of SchematicsModelField.

Had to take into consideration both versions of schematics, thus we check for the type of Error before trying to raise a new StereotypeConversionError.

alexandremartins98 committed:
Catch SchematicsConversionError in convert of SchematicsModelField

alexandremartins98 committed:
Remove not needed test helper

thisDash commented:
I am sorry for the code formatting changes, just unnecessary changes for you to pay attention to...

petee-d commented:
> I am sorry for the code formatting changes, just unnecessary changes for you to pay attention to...

Yeah, I was wondering when the time to start using Black would come... :D

petee-d committed:
Simplify raise_conversion_error

* made it private as it's probably not desired for users to import this
* used error.args[0] instead of error.messages as it was unclear why it
even exists for Schematics 2 DataError (I didn't see it defined
anywhere, yet it worked)
* simplify traversing the mapping until the first error sequence
* unify the way the first error message is obtained from the sequence
(the approach works for both Schematics versions, thus there's no need
for a split by version)
* documented the function

petee-d commented on stereotype/contrib/schematics.py:
```diff
@@ -27,10 +40,20 @@ class SchematicsModelField(ModelField):
-                 primitive_name: Optional[str] = Missing, to_primitive_name: Optional[str] = Missing):
-        super().__init__(default=default, hide_none=hide_none,
-                         primitive_name=primitive_name, to_primitive_name=to_primitive_name)
+    def __init__(
```
No need to change it in this PR, but I'd recommend not combining automated reformatting (with tools such as Black) in the same PR as other changes, and never in the same commit. Makes it quite difficult to find the actual changes between all the diffs from those tools. Plus it binds your changes together with the decision to adopt a particular formatting tool.

But again, not really an issue, I'll probably format all the stuff with Black eventually.

petee-d replied:
Ah, now I see you already talked about it in the description, please disregard. :)

petee-d commented on stereotype/contrib/schematics.py:
```diff
@@ -68,3 +104,33 @@ def _iterate_validation_errors(messages: dict) -> Iterable[PathErrorType]:
                 yield (str(key),) + path, error
+
+
+def raise_conversion_error(error):
```
I found this function quite difficult to understand, so I took the liberty of refactoring it, please see my commit's description for more info.
